### PR TITLE
Add seealso section to Windows and related modules

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This implements a `seealso:` section in the module documentation of Windows modules and related modules.

This relates to: #45949

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
Various modules

### CODE REVIEW

**Comment by dagwieers** [comment]
@acozine @gundalow @jborean93 Please don't let this one go stale.

File: lib/ansible/modules/files/synchronize.py [hunk 1]
```diff
@@ -185,7 +185,9 @@
      encounters an error. Those synchronizing large numbers of files that are willing to trade safety for performance should call rsync directly.
    - link_destination is subject to the same limitations as the underlaying rsync daemon. Hard links are only preserved if the relative subtrees
      of the source and destination are the same. Attempts to hardlink into a directory that is a subdirectory of the source will be prevented.
-
+seealso:
+- module: copy
+- module: win_robocopy
```
Comment (jborean93) [inline lib/ansible/modules/files/synchronize.py:190]: Not sure if we want to link `win_robocopy`. Technically it does do synchronization but that's only on remote to remote not local to remote.

File: lib/ansible/modules/windows/win_security_policy.py [hunk 2]
```diff
@@ -49,6 +39,16 @@
...
+notes:
+- This module uses the SecEdit.exe tool to configure the values, more details
+  of the areas and keys that can be configured can be found here
+  U(https://msdn.microsoft.com/en-us/library/bb742512.aspx).
+- If you are in a domain environment these policies may be set by a GPO policy,
+  this module can temporarily change these values but the GPO will override
+  it if the value differs.
+- You can also run C(SecEdit.exe /export /cfg C:\temp\output.ini) to view the
+  current policies set on your system.
+- When assigning user rights, use the M(win_user_right) module instead.
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_security_policy.py:51]: Probably want a seealso for `win_user_right`

**Review by jborean93 — COMMENTED** [review]
I'm good with the changes, just a few comments inline. I'll leave it up to the others to review and merge.

File: lib/ansible/modules/files/synchronize.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/files/synchronize.py:190]: So the **synchronize** module was mentioned in the **win_robocopy** module docs, and I tried to ensure we have a reverse link as well. I don't mind removing it though.

File: lib/ansible/modules/windows/win_security_policy.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_security_policy.py:51]: Good catch. Fixed!

File: lib/ansible/modules/files/synchronize.py (hunk 1 above)
Comment (jborean93) [reply lib/ansible/modules/files/synchronize.py:190]: I don't mind either way, as long as it's clear win_robocopy won't work for local to remote synchronization.

File: lib/ansible/modules/windows/win_dotnet_ngen.py [hunk 3]
```diff
@@ -25,9 +25,9 @@
     - There are in fact two scheduled tasks for ngen but they have no triggers so aren't a problem.
     - There's no way to test if they've been completed.
     - The stdout is quite likely to be several megabytes.
+options: {}
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_dotnet_ngen.py]: Please revert.

**Review by gundalow — COMMENTED** [review]
Thank you for adding this, a good addition though, please be mindful of git blame.

File: lib/ansible/modules/windows/win_regmerge.py [hunk 4]
```diff
@@ -46,6 +44,11 @@
      differences on subsequent runs.
      To force registry change, use M(win_regedit) with state=absent before
      using M(win_regmerge).
+seeals:
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_regmerge.py]: `seealso:`

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_dotnet_ngen.py (hunk 3 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_dotnet_ngen.py]: Should this not be the default if it does not exist ? An empty dictionary.

File: lib/ansible/modules/windows/win_regmerge.py (hunk 4 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_regmerge.py]: Fixed.

File: lib/ansible/modules/windows/win_dotnet_ngen.py (hunk 3 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_dotnet_ngen.py]: Fixed.

Comment (dagwieers) [reply lib/ansible/modules/windows/win_dotnet_ngen.py]: Ah, now I see that you object it being relocated?

Comment (gundalow) [reply lib/ansible/modules/windows/win_dotnet_ngen.py]: Yes, especially as there aren't any other changed being made to this file

File: lib/ansible/modules/windows/win_audit_policy_system.py [hunk 5]
```diff
@@ -37,6 +33,13 @@
     required: yes
     type: list
     choices: [ failure, none, success ]
+notes:
+  - It is recommended to take a backup of the policies before adjusting them for the first time.
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_audit_policy_system.py:37]: This sounds important, should it be in both places so people have a higher change of reading it?

File: lib/ansible/modules/windows/win_defrag.py [hunk 6]
```diff
@@ -16,6 +16,8 @@
 description:
 - Locates and consolidates fragmented files on local volumes to improve system performance.
 - 'More information regarding C(win_defrag) is available from: U(https://technet.microsoft.com/en-us/library/cc731650(v=ws.11).aspx)'
+requirements:
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_defrag.py:19]: Please revert this file

File: lib/ansible/modules/windows/win_disk_facts.py [hunk 7]
```diff
@@ -18,11 +18,11 @@
      its volumes and partitions if existent.
 requirements:
     - Windows 8.1 / Windows 2012 (NT 6.2)
-author:
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_disk_facts.py:21]: Please revert this file

File: lib/ansible/modules/windows/win_product_facts.py [hunk 8]
```diff
@@ -15,9 +15,9 @@
 description:
 - Provides Windows product information.
 version_added: '2.5'
+options: {}
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_product_facts.py:18]: Please revert this file

File: lib/ansible/modules/windows/win_rabbitmq_plugin.py [hunk 9]
```diff
@@ -15,8 +15,6 @@
 description:
   - Manage RabbitMQ plugins.
 version_added: "2.4"
-author:
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_rabbitmq_plugin.py:18]: Please revert this file

File: lib/ansible/modules/windows/win_snmp.py [hunk 10]
```diff
@@ -15,8 +15,6 @@
 module: win_snmp
 version_added: '2.8'
 short_description: Configures the Windows SNMP service
-author:
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_snmp.py:18]: Please revert this file

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_audit_policy_system.py (hunk 5 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_audit_policy_system.py:37]: The order in the python files do not reflect the order in the documentation or man page.
If you think notes should be before module options, we should fix that in the template.

File: lib/ansible/modules/windows/win_defrag.py (hunk 6 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_defrag.py:19]: But this is the order we use when generated. I think it helps readability if the order is aligned.
(Same for the previous remark where you wanted notes higher up, that should be reflected in the template)

**Comment by dagwieers** [comment]
@gundalow I have spend quite some time improving the documentation and I really am irritated that you want me to spend additional time over unimportant bits that absolutely make no difference except add consistency to the Windows modules. At this point I rather close the PR than spend more time on it.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merge once green

**Pull request merged**